src/Database/Bloodhound/Types.hs

```diff
@@ -542,3 +557,104 @@ instance (FromJSON a) => FromJSON (SearchResult a) where
 newtype ScrollId =
   ScrollId Text
   deriving (Eq, Show, Ord, ToJSON, FromJSON)
+
+newtype SearchTemplateId = SearchTemplateId Text deriving (Eq, Show)
+
+instance ToJSON SearchTemplateId where
+  toJSON (SearchTemplateId x) = toJSON x
+
+newtype SearchTemplateSource = SearchTemplateSource Text deriving (Eq, Show)
+
+instance ToJSON SearchTemplateSource where
+  toJSON (SearchTemplateSource x) = toJSON x
+
+instance FromJSON SearchTemplateSource where
+  parseJSON (String s) = pure $ SearchTemplateSource s
+  parseJSON _          = empty
+
+data ExpandWildcards = ExpandWildcardsAll
+  | ExpandWildcardsOpen
+  | ExpandWildcardsClosed
+  | ExpandWildcardsNone
+  deriving (Eq, Show)
+
+instance ToJSON ExpandWildcards where
+  toJSON ExpandWildcardsAll = String "all"
+  toJSON ExpandWildcardsOpen = String "open"
+  toJSON ExpandWildcardsClosed = String "closed"
+  toJSON ExpandWildcardsNone = String "none"
+
+instance FromJSON ExpandWildcards where
+  parseJSON (String "all") = pure $ ExpandWildcardsAll
+  parseJSON (String "open") = pure $ ExpandWildcardsOpen
+  parseJSON (String "closed") = pure $ ExpandWildcardsClosed
+  parseJSON (String "none") = pure $ ExpandWildcardsNone
+  parseJSON _          = empty
+
+data TimeUnits = TimeUnitDays 
+  | TimeUnitHours
+  | TimeUnitMinutes
+  | TimeUnitSeconds
+  | TimeUnitMilliseconds
+  | TimeUnitMicroseconds
+  | TimeUnitNanoseconds
+  deriving (Eq, Show)
+
+instance ToJSON TimeUnits where
+  toJSON TimeUnitDays = String "d"
+  toJSON TimeUnitHours = String "h"
+  toJSON TimeUnitMinutes = String "m"
+  toJSON TimeUnitSeconds = String "s"
+  toJSON TimeUnitMilliseconds = String "ms"
+  toJSON TimeUnitMicroseconds = String "micros"
+  toJSON TimeUnitNanoseconds = String "nanos"
+
+instance FromJSON TimeUnits where
+  parseJSON (String "d") = pure $ TimeUnitDays
+  parseJSON (String "h") = pure $ TimeUnitHours
+  parseJSON ( String "m") = pure $ TimeUnitMinutes
+  parseJSON (String "s") = pure $ TimeUnitSeconds
+  parseJSON (String "ms") = pure $ TimeUnitMilliseconds
+  parseJSON (String "micros") = pure $ TimeUnitMicroseconds
+  parseJSON (String "nanos") = pure $ TimeUnitNanoseconds
+  parseJSON _          = empty
+
+data SearchTemplate =
+  SearchTemplate {
```

Comment:
style nitpick: This is a bit different from the other data type definitions in this project. Would you mind changing it?